### ISSUE TITLE
IE conditional comments + domain name with ASCII encoding

### ISSUE THREAD

**Issue by Arttse** [validator/validator#439]
Check example:
```html
<!DOCTYPE html>
<html>
<head>
    <title>Test 1</title>
    <link rel="stylesheet" href="http://xn--d1abbgf6aiiy.xn--p1ai/css/style.css" type="text/css" media="all" />
    <!--[if lt IE 10]>
    <link rel="stylesheet" href="http://xn--d1abbgf6aiiy.xn--p1ai/css/ie.css" type="text/css" media="all" />
    <![endif]-->
    <!--[if lt IE 9]>
    <link rel="stylesheet" href="http://xn--d1abbgf6aiiy.xn--p1ai/css/ie8.css" type="text/css" media="all" />
    <![endif]-->
    <!--[if lt IE 8]>
    <link rel="stylesheet" href="http://xn--d1abbgf6aiiy.xn--p1ai/css/ie7.css" type="text/css" media="all" />
    <![endif]-->
</head>
<body>
    <p>Test</p>
</body>
</html>
```

Output:
```
Warning: The document is not mappable to XML 1.0 due to two consecutive hyphens in a comment.
```

Please correct me if I'm wrong.

**Comment by sideshowbarker** [comment]
That warning is intentional and correct.

The multiple dashes in those comments are not (any longer) and error in documents served as text/html, but they will still cause XML parsers to fail to parse the document at all.

So that’s why we emit that message, and that’s why it’s only a warning for the text/html case.

**Comment by Arttse** [comment]
Ok. As I understand, to avoid this warning, need to use a relative path?

**Comment by sideshowbarker** [comment]
> Ok. As I understand, to avoid this warning, need to use a relative path?

Well that would be one way, yeah.

But another way in this case in the general case of IDN domain names is just to use the real IDN domain name instead of punycode. No errors or warnings are reported for the following:

```html
<!DOCTYPE html>
<html>
<head>
    <title>Test 1</title>
    <link rel="stylesheet" href="http://президент.рф/css/style.css" type="text/css" media="all" />
    <!--[if lt IE 10]>
    <link rel="stylesheet" href="http://президент.рф/css/ie.css" type="text/css" media="all" />
    <![endif]-->
    <!--[if lt IE 9]>
    <link rel="stylesheet" href="http://президент.рф/css/ie8.css" type="text/css" media="all" />
    <![endif]-->
    <!--[if lt IE 8]>
    <link rel="stylesheet" href="http://президент.рф/css/ie7.css" type="text/css" media="all" />
    <![endif]-->
</head>
<body>
    <p>Test</p>
</body>
</html>
```

**Comment by Arttse** [comment]
Thanks for the replies :+1: It's important to me.
